homeassistant/components/homekit_controller/button.py

```diff
@@ -32,6 +32,10 @@
 class HomeKitButtonEntityDescription(ButtonEntityDescription):
     """Describes Homekit button."""
 
+    # HomeKitButton does not support UNDEFINED or None,
+    # restrict the type to str.
+    name: str = ""
```

Comment:
Should it be made compulsory using a mixin?

Reply:
Yeah, maybe.

I had an idea to do something like this:
```py
@dataclass
class NameRequiredMixin:
    """Mixin for required name."""

    def __post_init__(self) -> None:
        super().__post_init__()
        assert self.name


@dataclass
class AugustBinarySensorEntityDescription(BinarySensorEntityDescription, NameRequiredMixin):
    """Describes August binary_sensor entity."""
    # AugustBinarySensor does not support UNDEFINED or None, restrict the type to str.
    name: str = "
```